src/expertsystem/reaction/topology.py

```diff
@@ -571,8 +571,8 @@ def __post_init__(self) -> None:
         _assert_over_defined(self.topology.nodes, self.__node_props)
         _assert_over_defined(self.topology.edges, self.__edge_props)
 
-    def get_node_props(self, node_id: int) -> InteractionProperties:
-        return self.__node_props[node_id]
+    def get_node_props(self, node_id: int) -> Optional[InteractionProperties]:
+        return self.__node_props.get(node_id)
```

Comment:
Why did the `node_props` become optional now? The reason why i did not make this optional, is because I require those to exist. So that all of the other code working with the graphs can assume they exist

Reply:
If they should not be optional, we should check that all nodes have properties assigned. I remember that if I didn't use `dict.get` when running the notebooks with this new visualization procedure, the program would crash because of missing keys. But I would have to recheck; these commits have been extracted from a branch that had the work on the other PRs combined.

Reply:
> (...) the program would crash because of missing keys.

I checked, apparently the cause for that was something else. 05ef4cb reverts the `Optional` change.